modules/Matting/Global-Sampling-Method-for-Alpha-Matting/guidedfilter.cpp

```diff
@@ -0,0 +1,212 @@
+#include "guidedfilter.h"
+
+static cv::Mat boxfilter(const cv::Mat &I, int r)
+{
+    cv::Mat result;
+    cv::blur(I, result, cv::Size(r, r));
+    return result;
+}
+
+static cv::Mat convertTo(const cv::Mat &mat, int depth)
```

Comment:
These are functions that exist in OpenCV, why are we overwriting functionality? It will break compatibility and arrange unexpected behaviour.